### DISCUSSION TITLE
KlipperScreen and Python scripts

### DISCUSSION THREAD

**Discussion by DiegoHerrera1890** [KlipperScreen/KlipperScreen#1446] [Q&A]
Dear community,

I am running Klipper on my customized 3D printer and I need to implement an icon in KlipperScreen that, when clicked, can execute a Python script.

The Python script communicates with Moonraker to control the 3D printer and perform certain actions. However, I am facing an issue where KlipperScreen does not directly work with Python. Here's what I have tried so far:

1. Created an icon in KlipperScreen with the following configuration:

    ```ini
    [menu __main check_status]
    name: Check Ruso
    icon: status
    method: printer.gcode.script
    params: {"script":"SEND_MESSAGE Hello"}
    ```

2. Created a macro in `printer.cfg`:

    ```ini
    [gcode_macro SEND_MESSAGE]
    gcode:
        SYSTEM /home/diego/testing_code.py "{rawparams}"
    ```

However, I receive the following error:

```
[screen.py:show_popup_message()] - error: Malformed command 'SEND_MESSAGE Hello, Klipper!'
```

Does anyone have any suggestions on how to properly configure this? Any help would be greatly appreciated.

Thank you in advance!

**Comment by alfrix** [comment]
to run scripts from gcode macros you need to install the GCODE shell command extension, you can install it from KIAUH
